common/oatbox/cache/GcpTokenCacheItemPool.php

```diff
@@ -0,0 +1,251 @@
+
+        $this->log(__METHOD__, $keys);
+
+        return $this->getCache()->deleteItems($keys);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function save(CacheItemInterface $item)
+    {
+        $this->log(__METHOD__);
+
+        if ($this->isWritingDisabled()) {
+            return true;
+        }
+
+        $this->serializeToPersist($item);
+
+        return $this->getCache()->save($item);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function saveDeferred(CacheItemInterface $item)
+    {
+        $this->log(__METHOD__);
+
+        if ($this->isWritingDisabled()) {
+            return true;
+        }
+
+        $this->serializeToPersist($item);
+
+        return $this->getCache()->saveDeferred($item);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function commit()
+    {
+        $this->log(__METHOD__);
+
+        if ($this->isWritingDisabled()) {
+            return true;
+        }
+
+        return $this->getCache()->commit();
+    }
+
+    private function log(string $method, array $keys = []): void
+    {
+        if ($this->isDebug()) {
+            $this->getLogger()->debug(
+                sprintf(
+                    '[%s] Called method %s with key(s) %s',
+                    __CLASS__,
+                    $method,
+                    implode(',', $keys)
+                )
+            );
+        }
+    }
+
+    private function isDebug(): bool
+    {
+        if ($this->isDebug === null) {
+            $this->isDebug = (bool)$this->getOption(self::OPTION_ENABLE_DEBUG, false);
+        }
+
+        return $this->isDebug;
+    }
+
+    private function isWritingDisabled(): bool
+    {
+        if ($this->isWritingDisabled === null) {
+            $this->isWritingDisabled = (bool)$this->getOption(self::OPTION_DISABLE_WRITE, false);
+        }
+
+        return $this->isWritingDisabled;
+    }
+
+    private function getCache(): CacheItemPoolInterface
+    {
+        $this->cache = $this->createCache();
+
+        return $this->cache;
+    }
+
+    private function createCache(): CacheItemPoolInterface
+    {
+        return $this->getCacheFactory()->create(
+            [
+                CacheItemPoolFactory::CONFIG_PERSISTENCE => $this->getOption(self::OPTION_PERSISTENCE, 'redis'),
+            ]
+        );
+    }
+
+    private function unSerializeIfNecessary(CacheItemInterface $item): void
+    {
+        if (empty($item->get())) {
+            return;
+        }
+
+        $unSerialized = @unserialize($item->get());
+
+        if ($unSerialized !== false) {
+            $item->set($unSerialized);
+        }
+    }
+
+    private function serializeToPersist(CacheItemInterface $item): void
+    {
+        if (is_array($item->get())) {
+            $item->set(serialize($item->get()));
+        }
+    }
+
+    private function getCacheKey(): string
+    {
+        return (string)$this->getOption(self::OPTION_TOKEN_CACHE_KEY, 'GCP-TOKEN-SANCTUARY:GOOGLE_AUTH_PHP_GCE');
```

Comment:
Should we put this value into constant?

Reply:
I could, but would be only used here. Do you think it worth?